File: data/imported/knowledge/slides/python/merged/snippets/snippet_067.py
# Adapted from aspose.org: knowledge/slides/python/merged/snippets/snippet_067.py @ 7f72da4e1423546104b40fa8cebf5b9ae3ce9c91
# Imported under the authorization recorded in
# plans/investigations/evidence/imported-corpus-v1/licensing-resolution-state.md

def test_gradient_background(tmp_pptx):

    """Gradient background with tile flip round-trips."""

    pres = Presentation()

    slide = pres.slides[0]

    slide.background.type = BackgroundType.OWN_BACKGROUND

    slide.background.fill_format.fill_type = FillType.GRADIENT

    slide.background.fill_format.gradient_format.tile_flip = TileFlip.FLIP_BOTH



    pres2 = tmp_pptx(pres)

    bg = pres2.slides[0].background

    assert bg.type == BackgroundType.OWN_BACKGROUND

    assert bg.fill_format.fill_type == FillType.GRADIENT

    assert bg.fill_format.gradient_format.tile_flip == TileFlip.FLIP_BOTH

    pres2.dispose()

File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_067.cpp
TEST_F(SlidesIntegrationTest, IterateSlides) {
    Presentation pres;
    auto* layout = &pres.layout_slides()[0];
    pres.slides().add_empty_slide(layout);
    std::vector<Slide*> slides;
    for (auto& s : pres.slides()) {
        slides.push_back(s.get());
    }
    EXPECT_EQ(slides.size(), 2);
}